xdsl/parser/attribute_parser.py

```diff
@@ -504,21 +504,48 @@ def parse_shape_delimiter(self) -> None:
         # Move the lexer to the position after 'x'.
         self._resume_from(self._current_token.span.start + 1)
 
+    def _parse_empty_dimension_list(self) -> bool:
+        """
+        Parses an empty dimension list, which has it's own syntax:
+          empty-dimension-list ::= `[` `]`
+        Note that a regular dimension list has no square brackets.
+        """
+        if self.parse_optional_characters("["):
+            self.parse_characters("]")
+            return True
+
+        return False
```

Comment:
This should be done by the caller, not all dimension lists are printed with `[]`